packages/vite-plugin/src/node/plugin-contentScripts.ts

```diff
@@ -167,6 +167,8 @@ export const pluginContentScripts: CrxPluginFn = () => {
                 })
 
                 script.loaderName = this.getFileName(refId)
+              } else {
+                bundleFileInfo.code = `(function(){${bundleFileInfo.code}})()`
```

Comment:
We realized that the original loader which wrapped the content script in a function call had the side effect of variable scope isolation, which, if droppped, may result in a breaking change, if some variable names across different content scripts are the same (especially after vite mangles variable names). To preserve this behavior, we are wrapping here the content script in a (synchronous) self invoked function which should cause no issues since the script has no imports/exports

Reply:
now the comment is unclear
> // the loader uses IIFE which in this case needlessly
              // delays content script execution which may not be desired

because now the solution uses IIFE itself 
> // make sure the code is wrapped in a function invocation
   // to have the same scope isolation as the loader provides

``(function(){${bundleFileInfo.code}})()`` is a IIFE

Reply:
thanks, fixed 👍